File: backend/Dtos/response/PesajeResponseDTO.cs
using System.Text.Json.Serialization;
using backend.Enum;

namespace backend.Dtos.response;

public record PesajeResponseDTO(
    int Id_Pesaje,
    int Id_Paciente,
    decimal Peso_Kg,
    decimal? Porcentaje_Grasa,
    decimal? Masa_Muscular_Kg,
    DateTime Fecha_Pesaje,
    string? Nota,
    
    decimal Imc, 
    [property: JsonConverter(typeof(JsonStringEnumConverter))]
    EIMC ClasificacionImc,
    decimal DiferenciaPesoAnterior
);

File: frontend/types/dto/response/PesajeResponseDTO.ts
// src/types/dto/pesaje/PesajeResponseDTO.ts

export interface PesajeResponseDTO {
    id_Pesaje: number;
    id_Paciente: number;
    peso_Kg: number;
    porcentaje_Grasa?: number;
    masa_Muscular_Kg?: number;
    fecha_Pesaje: string;
    nota?: string;
    
    imc: number;
    clasificacionImc: string;
    diferenciaPesoAnterior: number;
}